src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void clearList() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void checkEmptyList_Ok() {
+        assertTrue(Storage.people.isEmpty());
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setPassword(validPassword);
+        user.setAge(validAge);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin(validLogin);
+        user.setAge(validAge);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin(validLogin);
+        user.setPassword("abcdf");
+        user.setAge(validAge);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setPassword("abc");
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("Dmitr");
+        user.setPassword(validPassword);
+        user.setAge(validAge);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setLogin("Dmi");
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_youngAge_notOk() {
+        User user = new User();
+        user.setLogin(validLogin);
+        user.setPassword(validPassword);
+        user.setAge(8);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setAge(-8);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin(validLogin);
+        user1.setPassword(validPassword);
+        user1.setAge(validAge);
+        Storage.people.add(user1);
+        User user2 = new User();
+        user2.setLogin(validLogin);
+        user2.setPassword(validPassword);
+        user2.setAge(validAge);
+        assertThrows(RegistationException.class, () -> {
+            registrationService.register(user2);
+        });
+    }
+
+    @Test
+    void register_user_Ok() throws RegistationException {
+        User user = new User();
+        user.setLogin(validLogin);
+        user.setPassword(validPassword);
+        user.setAge(validAge);
```

Comment:
I'd recommend to add this to ```@BeforeEach``` method so before test you have standard valid user.
Then in tests you may change only required field (password or login to invalid one, etc.)